### PR TITLE
[jenkins] feat: catapult code coverage job

### PR DESCRIPTION
**Pull request by Wayonb**
## What is the current behavior?
There is no code coverage job for catapult client

## What's the issue?
Currently code coverage for the catapult client is generated manually.

## How have you changed the behavior?
created a daily code coverage job for the catapult client.
It is base on gcc 10 until update all the test images

## How was this change tested?
Yes

### CODE REVIEW

File: jenkins/catapult/jenkins/catapult-client-build-catapult-project-daily.groovy [hunk 1]
```diff
@@ -73,6 +73,13 @@ pipeline {
 						}
 					}
 				}
+				stage('code coverage') {
+					steps {
+						script {
+							dispatch_build_job('gcc-10-code-coverage', 'tests-diagnostics')
```
Comment (Jaguar0625) [inline jenkins/catapult/jenkins/catapult-client-build-catapult-project-daily.groovy:79]: why tests-diagnostics? would probably not use that for cc build

Comment (Wayonb) [reply jenkins/catapult/jenkins/catapult-client-build-catapult-project-daily.groovy:79]: no specific reason.  will change to ``tests-metal``

File: jenkins/catapult/configuration.py [hunk 2]
```diff
@@ -31,9 +31,10 @@ def load_compiler_configuration(filepath):
 
 		sanitizers = configuration_yaml['sanitizers'].split(',') if 'sanitizers' in configuration_yaml else []
 		architecture = configuration_yaml['architecture']
+		enable_code_coverage = configuration_yaml['enable_code_coverage'] if 'enable_code_coverage' in configuration_yaml else False
```
[the remark this answers is not in the document]
Comment (Wayonb) [reply jenkins/catapult/configuration.py]: use the get() with a default value of False.  bool() seem to return True for any string which is not empty.  
``enable_code_coverage: nope`` would enable code coverage

**Review by Jaguar0625 — APPROVED** [review]

File: jenkins/catapult/runDockerBuildInnerBuild.py [hunk 3]
```diff
@@ -158,7 +159,7 @@ def main():
 
 	builder = BuildManager(args, process_manager, environment_manager)
 	env = LinuxEnvironment(builder.use_conan, process_manager, environment_manager)
-	env.prepare()
+	env.prepare(builder.enable_code_coverage, args.source_path)
```
Comment (gimre-xymcity) [inline jenkins/catapult/runDockerBuildInnerBuild.py:162]: why is dest dir `source_path/_build`? 😬 
(isn't it possible to make build inside /tmp, like earlier?)

**Review by gimre-xymcity — APPROVED** [review]

**Pull request merged**